packages/lesswrong/server/repos/PostsRepo.ts

```diff
@@ -412,6 +412,22 @@ class PostsRepo extends AbstractRepo<"Posts"> {
       WHERE contents->>'html' LIKE '%elicit-binary-prediction%'
     `);
   }
+
+  // Used in the cronjob for archiving stale dialogues and notifying users about them 
+  async getStaleDialogues(cutOffTime:Date): Promise<DbPost[]> {
+    return this.getRawDb().any(`
+      -- PostsRepo.getStaleDialogues
```

Comment:
I have an index in https://github.com/ForumMagnum/ForumMagnum/pull/8551 which should improve the performance of this query significantly.

EXPLAIN ANALYZE against prod (which doesn't have the new index yet)
```
"Group  (cost=35822.98..35825.30 rows=75 width=4398) (actual time=1242.225..1242.725 rows=101 loops=1)"
"  Group Key: p._id"
"  ->  Sort  (cost=35822.98..35824.14 rows=464 width=4398) (actual time=1242.215..1242.413 rows=1915 loops=1)"
"        Sort Key: p._id"
"        Sort Method: quicksort  Memory: 3180kB"
"        ->  Nested Loop  (cost=0.43..35802.43 rows=464 width=4398) (actual time=4.828..1234.539 rows=1915 loops=1)"
"              ->  Seq Scan on ""Posts"" p  (cost=0.00..35308.49 rows=75 width=4398) (actual time=2.502..692.633 rows=151 loops=1)"
"                    Filter: ((draft IS TRUE) AND (""collabEditorDialogue"" IS TRUE) AND (""deletedDraft"" IS NOT TRUE))"
"                    Rows Removed by Filter: 95147"
"              ->  Index Only Scan using ""idx_Revisions_documentId_version_fieldName_editedAt"" on ""Revisions"" r  (cost=0.43..6.53 rows=6 width=18) (actual time=1.227..3.571 rows=13 loops=151)"
"                    Index Cond: ((""documentId"" = (p._id)::text) AND (""editedAt"" < (CURRENT_TIMESTAMP - '24 days'::interval)))"
"                    Heap Fetches: 1356"
"Planning Time: 58.315 ms"
"Execution Time: 1242.919 ms"
```

EXPLAIN ANALYZE against dev (where does have the new index)
```
"HashAggregate  (cost=1790.45..1791.50 rows=105 width=4392) (actual time=7.051..7.318 rows=247 loops=1)"
"  Group Key: p._id"
"  Batches: 1  Memory Usage: 583kB"
"  ->  Nested Loop  (cost=10.20..1788.58 rows=748 width=4392) (actual time=0.099..6.078 rows=1356 loops=1)"
"        ->  Bitmap Heap Scan on ""Posts"" p  (cost=9.76..1301.11 rows=105 width=4392) (actual time=0.063..0.687 rows=268 loops=1)"
"              Recheck Cond: (""collabEditorDialogue"" IS TRUE)"
"              Filter: ((draft IS TRUE) AND (""deletedDraft"" IS NOT TRUE))"
"              Rows Removed by Filter: 52"
"              Heap Blocks: exact=294"
"              ->  Bitmap Index Scan on ""idx_Posts_max_postedAt_mostRecentPublishedDialogueResponseDate""  (cost=0.00..9.74 rows=351 width=0) (actual time=0.026..0.026 rows=419 loops=1)"
"        ->  Index Only Scan using ""idx_Revisions_documentId_version_fieldName_editedAt"" on ""Revisions"" r  (cost=0.43..4.57 rows=7 width=18) (actual time=0.007..0.018 rows=5 loops=268)"
"              Index Cond: ((""documentId"" = (p._id)::text) AND (""editedAt"" < (CURRENT_TIMESTAMP - '24 days'::interval)))"
"              Heap Fetches: 0"
"Planning Time: 0.941 ms"
"Execution Time: 7.409 ms"
```